src/baldr/graphtile.cc

```diff
@@ -535,6 +535,11 @@ const TransitDeparture* GraphTile::GetNextDeparture(const uint32_t lineid,
     return nullptr;
   }
 
+  // adjust the time if it is after midnight.
+  uint32_t current = current_time;
+  while (current >= 86400)
+    current -= 86400;
+
```

Comment:
What happens if current time is just before midnight? With a binary search it will find departures at the end of the day? How does this find any after midnight departures? Wouldn't this force walking to another station so current time is after midnight so it can then find a departure?